docs: adding docs for online algorithms (#2615)
oneapi-src/oneDAL: docs/source/onedal/algorithms/covariance/covariance.rst

the means is a :math:`1 \times p` matrix, the covariance and the correlation matrices are :math:`p \times p` square matrices.
The means, the covariance, and the correlation are computed with the following formulas:

.. list-table::
   :widths: 20 50
   :header-rows: 1
   :align: left

   * - Statistic
     - Definition
   * - Means
     - :math:`M = (m_{1}, \ldots , m_{p})`, where :math:`m_{j}=\frac{1}{n}\sum _{i}{x}_{ij}`
   * - Covariance matrix (sample)
     - :math:`Cov = (v_{ij})`, where :math:`v_{ij}=\frac{1}{n-1}\sum_{k=1}^{n}(x_{ki}-m_{i})(x_{kj}-m{j})`, :math:`i=\overline{1,p}`, :math:`j=\overline{1,p}`
   * - Covariance matrix (maximum likelihood)

>      - :math:`Cov' = (v'_{ij})`, where :math:`v'_{ij}=\frac{1}{n}\sum_{k=1}^{n}(x_{ki}-m_{i})(x_{kj}-m{j})`, :math:`i=\overline{1,p}`, :math:`j=\overline{1,p}`

Reviewer: I think this equation is not needed here because we always compute unbiased sample covariance matrix as far as I understand.

Author: No, we support biased and unbiased since https://github.com/oneapi-src/oneDAL/pull/2581

   * - Correlation matrix
     - :math:`Cor = (c_{ij})`, where :math:`c_{ij}=\frac{v_{ij}}{\sqrt{v_{ii}\cdot v_{jj}}}`, :math:`i=\overline{1,p}`, :math:`j=\overline{1,p}`

.. _covariance_p_math:

Partial Computing
-----------------

> Given a block of a dataset :math:`X = \{ x_1, \ldots, x_n \}` with :math:`n` feature vectors of dimension :math:`p`,

Reviewer 2: ```suggestion
Given a block of a :math:`X = \{ x_1, \ldots, x_n \}` dataset with :math:`n` feature vectors of :math:`p` dimension,
```

> the sums is a :math:`1 \times p` matrix, the crossproduct is :math:`p \times p` square matrices.

Reviewer 2: ```suggestion
the sums is a :math:`1 \times p` matrix, the cross product is :math:`p \times p` square matrices.
```

> The sums, the crossproduct are computed with the following formulas:

Reviewer 2: ```suggestion
The sums and cross product are computed with the following formulas:
```

.. list-table::
   :widths: 20 50
   :header-rows: 1
   :align: left

   * - Statistic
     - Definition
   * - Sums
     - :math:`S = (m_{1}, \ldots , m_{p})`, where :math:`m_{j}=sum _{i}{x}_{ij}`

>    * - Crossproduct matrix

Reviewer 2: ```suggestion
   * - Cross product matrix
```

>      - :math:`Crossproduct = (v_{ij})`

Reviewer: ```suggestion
     - :math:`Crossproduct = (v_{ij}) = \sum_{k=1}^{n}(x_{ki}-m_{i})(x_{kj}-m{j})`, :math:`i=\overline{1,p}`, :math:`j=\overline{1,p}`
```

I think also that v_ij needs to be changed to some other name because v_ij is already used in covariance definition.

.. _covariance_f_math:

Finalize Computing
------------------

Given a partial result with partial products,

> the means is a :math:`1 \times p` matrix, the covariance and the correlation matrices are :math:`p \times p` square matrices.

Reviewer 2: ```suggestion
the mean is a :math:`1 \times p` matrix, the covariance and correlation matrices are :math:`p \times p` square matrices.
```

Author: Does it make sense to use means, due to here, it means a vector of mean values?

> The means, the covariance, and the correlation are computed with the following formulas:

Reviewer 2: ```suggestion
The mean, the covariance, and the correlation are computed with the following formulas:
```
